src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_ReadOnlyStruct.cs

```diff
@@ -0,0 +1,675 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using System.Collections.Immutable;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    [CompilerTrait(CompilerFeature.ReadOnlyReferences)]
+    public class ReadOnlyStruct : CSharpTestBase
```

Comment:
It is more of a test for embedded attributes in general, so it was covered in the tests when fists such scenario was introduced. I am not sure we need to redo similar tests for other embedded attributes or, as in this case, when we use the same embedded attribute in one more scenario. 
It would not have much to do with actual readonly structs as a feature.

---
In reply to: [119931556](https://github.com/dotnet/roslyn/pull/19888#discussion_r119931556) [](ancestors = 119931556,119918933)